src/Controller/Admin/Asset/AssetHelperController.php

```diff
@@ -340,6 +339,18 @@ protected function getFieldGridConfig(array $field, string $language = '', strin
             $result['layout']['fieldtype'] = 'manyToOneRelation';
             $result['layout']['subtype'] = $type;
         }
+        if(!$predefined && $type != 'system') {
+            $eventDispatcher = \Pimcore::getKernel()->getContainer()->get('event_dispatcher');
+            $assetGetFieldGridConfig = new GenericEvent($this, [
+                'field' => $field,
+                'result' => $result
+            ]);
+
+            $eventDispatcher->dispatch($assetGetFieldGridConfig, AdminEvents::ASSET_GET_FIELD_GRID_CONFIG);
```

Comment:
why dispatch only for non-predefined & non-system?

Reply:
To make sure we are only dispatching it for those which belongs to asset meta data class definition

Reply:
It should be generic so as can be used anywhere not just asset-metadata

Reply:
Done in [8f5dc16](https://github.com/pimcore/admin-ui-classic-bundle/pull/271/commits/8f5dc165273d808b1a6dbbcdb5aa631a5eeae897)